packages/muelu/src/MueCentral/MueLu_VariableContainer.hpp

```diff
@@ -243,31 +251,45 @@ namespace MueLu {
       typedef Teuchos::RCP<Operator> TO;
       typedef Teuchos::RCP<Matrix>   TM;
 
-      const std::string typeTOName = Teuchos::TypeNameTraits<TO>::name();
-      const std::string typeTMName = Teuchos::TypeNameTraits<TM>::name();
-      TEUCHOS_TEST_FOR_EXCEPTION(data_ == NULL, Teuchos::bad_any_cast,
-                                 "Error, cast to type Data<" << typeTOName << "> failed since the content is NULL");
+      if (data_ == NULL) // NVR added guard to avoid unnecessary call to TypeNameTraits<TO>::name(), which calls Teuchos::demangleName(), which can be expensive
```

Comment:
Can you clean this up a bit further, similar to the the above `get` call? Change this `if` to
```c++
if (data_ == NULL && data_->type() != typeid(TO) && data_->type() != typeid(TM)) {
 // throw appropriate exceptions
}
```
then deal with `TO`/`TM`.